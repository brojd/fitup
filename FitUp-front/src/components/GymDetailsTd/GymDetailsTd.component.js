import React, { PropTypes } from 'react';
import styles from './GymDetailsTd.stylesheet.css';
import Image from '../Image/Image.component';
import classNames from 'classnames';

const GymsDetailsTd = (props) => ((
  <td  className={classNames('uk-width-2-10', styles.gymTable__td)}>
    <ul className='uk-list'>
      <h4 className={styles.gymTable__heading}>{props.gym.name}</h4>
      <Image className='uk-float-right uk-width-3-10' src={props.gym.logo_url} alt='gym_logo' />
      <li className='uk-width-4-10 uk-float-left'>{props.gym.address.street} {props.gym.address.number},</li>
      <li className='uk-width-4-10 uk-float-left'>{props.gym.address.city}</li>
      <div className='uk-clearfix'></div>
    </ul>
    <ul className='uk-list'>
      <h4 className={styles.gymTable__heading}>Pricing:</h4>
      <li>
        {props.gym.pricing.map((price) => (
          <div key={price.name}>{price.name}: {price.price} PLN</div>
        ))}
      </li>
    </ul>
    <ul className='uk-list'>
      <h4 className={styles.gymTable__heading}>Cards accepted:</h4>
      <li>
        {props.gym.cards_accepted.map((card, i) => {
          if (card.accepted) {
            return (
              <div className='uk-badge uk-margin-small-right uk-margin-small-bottom uk-width-3-10'
                   key={i}>
                {card.name}
              </div>
            );
          }})
        }
      </li>
    </ul>
  </td>
));

GymsDetailsTd.propTypes = {
  gym: PropTypes.object
};

export default GymsDetailsTd;
