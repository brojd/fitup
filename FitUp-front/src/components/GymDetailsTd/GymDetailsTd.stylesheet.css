.gymTable__heading {
  border-bottom: 1px solid rgba(80, 0, 0, 0.26); }

.gymTable__td {
  padding: 21px 21px 21px 0 !important; }

/*# sourceMappingURL=GymDetailsTd.stylesheet.css.map */
