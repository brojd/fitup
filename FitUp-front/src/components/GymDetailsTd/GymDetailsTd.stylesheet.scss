@import "../../../sass_variables/colors";
@import "../../../sass_variables/widths";

.gymTable__heading {
  border-bottom: 1px solid rgba(80, 0, 0, 0.26);
}

.gymTable__td {
  padding: 3*$whitespace 3*$whitespace 3*$whitespace 0 !important;
}
