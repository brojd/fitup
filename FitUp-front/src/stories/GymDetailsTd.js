import React from 'react';
import { storiesOf} from '@kadira/storybook';
import GymDetailsTd from '../components/GymDetailsTd/GymDetailsTd.component';
import GYMS from './mock_data/gyms';

storiesOf('GymDetailsTd', module)
  .add('GymDetailsTd component', () => (
    <div className='uk-width-2-10'>
      <table className='uk-table'>
        <tbody>
          <tr>
            <GymDetailsTd gym={GYMS[0]}/>
          </tr>
        </tbody>
      </table>
    </div>
  ));
