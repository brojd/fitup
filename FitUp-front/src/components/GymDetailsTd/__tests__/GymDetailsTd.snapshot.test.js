import React from 'react';
import GymDetailsTd from '../GymDetailsTd.component';
import renderer from 'react-test-renderer';

test('GymDetailsTd renders correctly with given gym property', () => {
  
  const GYM = {
    'email': 'gym1@gym1.pl',
    'password': 'gym1',
    'name': 'Fitness Academy',
    'address': {'city': 'Wrocław', 'postal_code': '50-001', 'street':'Długa', 'number':3},
    'description': 'Lorem ipsum in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    'logo_url': 'https://pbs.twimg.com/profile_images/378800000513930765/1518318ade4947d639fca13c47b457f6.jpeg',
    'photo_urls': [
      'https://www.thegymgroup.com/fileadmin/uploads/gym/Photos/Gym_Leicester/Leicester14.jpg'
    ],
    'pricing': [
      {'name':'1 workout', 'price':15},
      {'name':'1 hour', 'price':40},
      {'name':'1 month', 'price':130}
    ],
    'cards_accepted': [
      {'name':'Multisport', 'accepted':true},
      {'name':'ActiveSport', 'accepted':true},
      {'name':'OKsystem', 'accepted':true},
      {'name':'FitWro', 'accepted':false},
      {'name':'FitProfit', 'accepted':false}
    ]
  };
  
  const component = renderer.create(
    <GymDetailsTd gym={GYM} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
