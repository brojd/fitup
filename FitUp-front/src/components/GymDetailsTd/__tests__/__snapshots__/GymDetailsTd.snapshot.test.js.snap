exports[`test GymDetailsTd renders correctly with given gym property 1`] = `
<td
  className="uk-width-2-10 gymTable__td">
  <ul
    className="uk-list">
    <h4
      className="gymTable__heading">
      Fitness Academy
    </h4>
    <img
      alt="gym_logo"
      className="uk-float-right uk-width-3-10"
      height={150}
      onClick={undefined}
      src="https://pbs.twimg.com/profile_images/378800000513930765/1518318ade4947d639fca13c47b457f6.jpeg"
      width={150} />
    <li
      className="uk-width-4-10 uk-float-left">
      Długa
       
      3
      ,
    </li>
    <li
      className="uk-width-4-10 uk-float-left">
      Wrocław
    </li>
    <div
      className="uk-clearfix" />
  </ul>
  <ul
    className="uk-list">
    <h4
      className="gymTable__heading">
      Pricing:
    </h4>
    <li>
      <div>
        1 workout
        : 
        15
         PLN
      </div>
      <div>
        1 hour
        : 
        40
         PLN
      </div>
      <div>
        1 month
        : 
        130
         PLN
      </div>
    </li>
  </ul>
  <ul
    className="uk-list">
    <h4
      className="gymTable__heading">
      Cards accepted:
    </h4>
    <li>
      <div
        className="uk-badge uk-margin-small-right uk-margin-small-bottom uk-width-3-10">
        Multisport
      </div>
      <div
        className="uk-badge uk-margin-small-right uk-margin-small-bottom uk-width-3-10">
        ActiveSport
      </div>
      <div
        className="uk-badge uk-margin-small-right uk-margin-small-bottom uk-width-3-10">
        OKsystem
      </div>
    </li>
  </ul>
</td>
`;
